### PR TITLE
Add webhook statistics to the header of the Incoming Webhooks page

### LINKED ISSUE #44000: Add errors/no events numbers to list webhooks UI
According to [mockups and discussion with Rob](https://www.figma.com/file/ngPEwLqRVFAfXQC7lE0WLz/Webhooks?node-id=1%3A4598), we need to include 2 counters for webhooks with errors and without received events.

/cc @jplahn @ryphil

### PR DESCRIPTION
**Pull request by pjlast**
Closes #44000 

Adds stats to the header of the Incoming webhooks page.

image

## Test plan

Storybook updated, tests added for GraphQL resolver updates.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.89 kb) | 0.02% (+1.89 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/pjlast/44000-webhooks-errors-or-no-events-counter/compare-report.html;tab=live_object) for a full comparison between the commits a81ffa9 and 443252d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/site-admin/backend.ts [hunk 1]
```diff
@@ -1002,6 +1004,34 @@ export const DELETE_WEBHOOK = gql`
...
+
+export const useWebhookPageHeader = (): { loading: boolean; totalErrors: number; totalNoEvents: number } => {
+    const { data, loading } = useQuery<WebhookPageHeaderResult, WebhookPageHeaderVariables>(WEBHOOK_PAGE_HEADER, {})
+    const totalNoEvents =
+        data?.webhooks.nodes.reduce((numNoEvents, webhook) => {
+            if (webhook.webhookLogs?.totalCount === 0) {
+                return numNoEvents + 1
+            }
+            return numNoEvents
+        }, 0) || 0
```
Comment (mrnugget) [inline client/web/src/site-admin/backend.ts:1030]:
```suggestion
        data?.webhooks.nodes.filter(webhook => webhook.webhookLogs?.totalCount === 0).length
```

No?

File: enterprise/cmd/frontend/internal/repos/webhooks/resolvers/resolver_test.go [hunk 2]
```diff
@@ -222,6 +264,110 @@ func TestListWebhooks(t *testing.T) {
...
+                            "webhookLogs": {
+                                "totalCount": 0
+                            }
+
+						}
+					],
+					"totalCount":4,
+					"pageInfo":{"hasNextPage":false}
+				}}`,
+		},
```
Comment (mrnugget) [inline enterprise/cmd/frontend/internal/repos/webhooks/resolvers/resolver_test.go:318]: The formatting here is really weird. Can you make these single lines per webhook maybe?

File: enterprise/cmd/frontend/internal/repos/webhooks/resolvers/resolver_test.go [hunk 3]
```diff
@@ -222,6 +264,110 @@ func TestListWebhooks(t *testing.T) {
...
+							id
+                            webhookLogs {
+                                totalCount
+                            }
+						}
+						totalCount
+						pageInfo { hasNextPage }
+					}
+				}
+			`,
```
Comment (mrnugget) [inline enterprise/cmd/frontend/internal/repos/webhooks/resolvers/resolver_test.go:284]: Mixed tabs and spaces in here

File: enterprise/cmd/frontend/internal/repos/webhooks/resolvers/resolver_test.go [hunk 4]
```diff
@@ -222,6 +264,110 @@ func TestListWebhooks(t *testing.T) {
...
+                            "webhookLogs": {
+                                "totalCount": 0
+                            }
+
+						}
+					],
+					"totalCount":4,
+					"pageInfo":{"hasNextPage":false}
+				}}`,
+		},
```
Comment (mrnugget) [inline enterprise/cmd/frontend/internal/repos/webhooks/resolvers/resolver_test.go:370]: Same as above

**Review by mrnugget — APPROVED** [review]

File: client/web/src/site-admin/SiteAdminWebhooksPage.module.scss [hunk 5]
```diff
@@ -5,3 +5,18 @@
         align-items: center;
     }
 }
+
+.grid {
+    display: grid;
+    grid-template-columns: max-content max-content 1fr max-content;
```
Comment (sashaostrikov) [inline client/web/src/site-admin/SiteAdminWebhooksPage.module.scss:11]: looks like you don't really need 4 columns here? only 2 are used. or is it made to shift these 2 `PerformanceGauge`s to the left?

File: client/web/src/site-admin/backend.ts (hunk 1 above)
Comment (sashaostrikov) [reply client/web/src/site-admin/backend.ts:1030]: `filter` looks cleaner IMHO

File: client/web/src/site-admin/SiteAdminWebhooksPage.module.scss (hunk 5 above)
Comment (pjlast) [reply client/web/src/site-admin/SiteAdminWebhooksPage.module.scss]: No that was an accidental leftover from the old webhooks page. Thanks! removed

**Review by ryanslade — APPROVED** [review]
LGTM. 

Stepping back though, wouldn't it be more useful to have these per row? Seeing the summary at the top doesn't help figure out which webhook is having errors. 

Fine to leave it as is for now, but worth discussing I think.

**Pull request merged**